extension/threadpool/threadpool.h

```diff
@@ -14,6 +14,32 @@
 
 #include <pthreadpool.h>
 
+/*
+ * Threadpool Options:
+ *
+ * Threadpool size has a sizble affect on performance. The following
+ * options are exposed to control the threadpool size.
+ *
+ * EXECUTORCH_THREADPOOL_SIZE: int - Set the size of the threadpool,
+ * in number of threads.
+ *
+ *  Special Values:
+ *   -  0: Use a perforance heuristic to determine the default size,
+ *          based on the active hardware. This is the default mode
+ *          for CMake.
+ *   - -1: Set the thread count equal to the number of cores on the
+ *          active hardware.
+ *
+ *   Any other positive value will be interpreted as a thread count.
+ *   For example, setting EXECUTORCH_THREADPOOL_SIZE=4 will default
+ *   the threadpool to use 4 threads.
+ */
+
+#ifndef EXECUTORCH_THREADPOOL_SIZE
```

Comment:
why is this a compile time option? How do you expect users to use it?

Reply:
I've piped it through CMake options now. In theory, I'd prefer to have a runtime API, but I don't really see a good way to do this given that it needs to be set prior to threadpool creation, which is implicit. I'm open to ideas. My expectation is that 99.9% of users will use the default perf core option, so needing to override at build time is an acceptable compromise.